src/transfers-history/adapters/web3/SpokePoolEventsQuerier.ts

```diff
@@ -78,27 +78,40 @@ export class SpokePoolEventsQuerier implements ISpokePoolContractEventsQuerier {
     let retryWithLowerBatchSize;
 
     do {
+      const blockRangeSizeAtStart = this.blockRangeSize;
       try {
         retryWithLowerBatchSize = false;
         events = [];
 
         if (this.blockRangeSize) {
-          for (const [intervalStart, intervalEnd] of getSamplesBetween(from, to, this.blockRangeSize)) {
-            const newEvents = await this.spokePool.queryFilter(filters, intervalStart, intervalEnd);
-            events.push(...newEvents);
-          }
+          const intervals = getSamplesBetween(from, to, this.blockRangeSize);
+          // query events only for the first interval to make sure block range is fine
+          const [intervalStart, intervalEnd] = intervals[0];
```

Comment:
theres an assumption here that you will only retry if the range is too large, and that you will know if the range is too large on the first query.  seems like you can simplify the code somewhat by testing the first query, adjusting to correct block range, and then requesting the rest of the ranges without having to add any retry logic for the bulk of the queries. i dont think this materially affects the performance/logic, but might make the code a bit cleaner

Reply:
I wanted to make a change, but I've just remembered that the same error code could be thrown if the length of the response is greater than a particular value. So actually all block ranges have to monitored for this particular error